mne/cov.py

```diff
@@ -956,12 +928,15 @@ def _compute_covariance_auto(data, method, info, method_params, cv,
                              scalings, n_jobs, stop_early, picks_list, rank):
     """Compute covariance auto mode."""
     # rescale to improve numerical stability
+    from .io import RawArray
+    raw_temp = RawArray(data.T, info)
+    if rank != 'full':
```

Comment:
why this branching?

Reply:
It saves a bit of time not to do the multiplications with `np.eye`, but I'll remove it because I agree it makes the code clearer